addons/test_mail/tests/test_message_post.py

```diff
@@ -36,6 +41,19 @@ def test_message_post_return(self):
         messageId = call_kw(self.env['mail.channel'], 'message_post', [test_channel.id], {'body': 'test'})
         self.assertTrue(isinstance(messageId, int))
 
+    @users('demo')
+    def test_mail_add_signature(self):
+        user = self.env.user
+        self.test_track.user_id = user
```

Comment:
You may also browse the record as current user, otherwise it is still browsed as superuser (test_record = self.test_record.with_user(self.env.user)).

Reply:
I tried to use self.test_track.with_user(self.env.user).message_post but it looked like it wasn't replacing the user_id in the record and therefore no signature was added so I kept it like that. Maybe I made a mistake